services/AttendanceService.ts

```diff
@@ -48,38 +48,72 @@ export default class AttendanceService {
   }
 
   private async writeEventAttendance(user: UserModel, event: EventModel, asStaff: boolean,
-    txn: EntityManager, description?: string): Promise<AttendanceModel> {
+    txn: EntityManager): Promise<AttendanceModel> {
     const attendedAsStaff = asStaff && user.isStaff() && event.requiresStaff;
     const pointsEarned = attendedAsStaff ? event.pointValue + event.staffPointBonus : event.pointValue;
 
     await Repositories.activity(txn).logActivity(
       user,
       attendedAsStaff ? ActivityType.ATTEND_EVENT_AS_STAFF : ActivityType.ATTEND_EVENT,
       pointsEarned,
-      description,
     );
     await Repositories.user(txn).addPoints(user, pointsEarned);
 
     return Repositories.attendance(txn).attendEvent(user, event, attendedAsStaff);
   }
 
-  public async submitAttendanceForUser(userUuid: Uuid, eventUuid: Uuid, asStaff = false,
-    admin: UserModel): Promise<PublicAttendance> {
+  public async submitAttendanceForUsers(emails: string[], eventUuid: Uuid, asStaff = false,
+    onBehalfOfUser: UserModel): Promise<PublicAttendance[]> {
     return this.transactions.readWrite(async (txn) => {
       const event = await Repositories.event(txn).findByUuid(eventUuid);
       if (!event) throw new NotFoundError('This event doesn\'t exist');
 
-      const user = await Repositories.user(txn).findByUuid(userUuid);
-      if (!user) throw new NotFoundError('This user was not found');
+      const users = await Repositories.user(txn).findByEmails(emails);
+      const emailsFound = users.map((user) => user.email);
+      const emailsNotFound = emails.filter((email) => !emailsFound.includes(email));
 
-      const attendanceRepository = Repositories.attendance(txn);
-      const hasAlreadyAttended = await attendanceRepository.hasUserAttendedEvent(user, event);
-      if (hasAlreadyAttended) throw new UserError('This user has already attended this event');
+      if (emailsNotFound.length > 0) {
+        throw new BadRequestError(`Couldn't find accounts matching these emails: ${emailsNotFound}`);
```

Comment:
Yeah makes sense, but we'd need to define specific error types / constructors for that in a different PR like you mentioned since we can't pass in a body to `HttpError` (it just accepts a `message`)